src/lib/Lib/MrDocsCompilationDatabase.cpp

```diff
@@ -165,6 +167,40 @@ adjustCommandLine(
     return new_cmdline;
 }
 
+std::optional<std::string> 
+executeCmakeExportCompileCommands(llvm::StringRef cmakePath, llvm::StringRef cmakeListsPath) 
+{
+    if ( ! llvm::sys::fs::exists(cmakePath)) {
+        return std::nullopt;
+    }    
+    if ( ! llvm::sys::fs::exists(cmakeListsPath)) {
+        return std::nullopt;
+    }
+
+    llvm::SmallString<128> stdOutPath;
+    if (auto ec = llvm::sys::fs::createTemporaryFile("stdout", "txt", stdOutPath)) 
+    {
+        return std::nullopt;
+    }
+
+    llvm::SmallString<128> stdErrPath;    
+    if (auto ec = llvm::sys::fs::createTemporaryFile("stderr", "txt", stdErrPath)) 
+    {
+        return std::nullopt;
+    }
+
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), stdOutPath.str(), stdErrPath.str()};
+    std::vector<llvm::StringRef> const args = {cmakePath, cmakeListsPath, "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"};
+    int const result = llvm::sys::ExecuteAndWait(cmakePath, args, std::nullopt, redirects);
+
+    if (result != 0) 
+    {
+        return std::nullopt;
+    }
+
+    return "./compile_commands.json";
```

Comment:
What's the point of always returning the same string?

Reply:
You're right about the function currently always returning the same string. My intention was to encapsulate the responsibility of determining the path to "compile_commands.json" within this function, so the caller doesn't need to handle it. However, I agree that this implementation isn't entirely satisfactory.

Additionally, I'm still exploring whether it's feasible to generate "compile_commands.json" in a directory other than the working directory. I plan to address this soon and will add a TODO note to remind us of this pending improvement.

Reply:
It definitely *has* to be another directory. We can't mess with other people's source directories. It has to be in a build directory and a temporary build directory unless the user provides some directory considered safe.